src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  demage(amout) {
+    this.health -= amout;
+
+    if (this.health <= 0) {
+      const indexOfThis = Animal.alive.indexOf(this);
+
+      Animal.alive.splice(indexOfThis, 1);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(obj) {
+    if (obj instanceof Carnivore || obj.hidden) {
+      return;
+    }
```

Comment:
[FUNCTION NAMING] - The condition in the `bite` method incorrectly checks if the object is an instance of `Carnivore` or if it is hidden. According to the task description, the method should only fail to work if the object is hidden. The check for `Carnivore` instance is not necessary because the `bite` method should only take a `Herbivore` object as a parameter.